android/src/com/mapswithme/util/UiUtils.java

```diff
@@ -363,72 +357,100 @@ public static boolean isViewTouched(@NonNull MotionEvent event, @NonNull View vi
     return viewRect.contains(x, y);
   }
 
-  public static int getStatusBarHeight(@NonNull Context context)
+  public static void setupTransparentStatusBar(@NonNull Activity activity)
   {
-    int result = 0;
-    Resources res = context.getResources();
-    int resourceId = res.getIdentifier("status_bar_height", "dimen", "android");
-    if (resourceId > 0)
-      result = res.getDimensionPixelSize(resourceId);
+    Window window = activity.getWindow();
+    window.getDecorView().setFitsSystemWindows(false);
+    window.setStatusBarColor(Color.TRANSPARENT);
+  }
 
-    return result;
+  public static void setLightStatusBar(@NonNull Activity activity, boolean isLight)
+  {
+    Window window = activity.getWindow();
+    View decorView = window.getDecorView();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+    {
+      WindowInsetsControllerCompat wic = WindowCompat.getInsetsController(window, decorView);
+      if (wic != null)
```

Comment:
Why it can be null here? Is it the best strategy to do nothing if it's null?

Reply:
`getInsetsController` can return null if the provided view is not attached to a window or has no decor view in its tree. But as I pass the window's decor view this should not happen. But then the function hints the compiler that the value could be null so not checking will raise a warning.

Reply:
In this case, it is way better to explicitly disable the warning and add a comment, that it should not be null here. And crash if it's null. And then fix it properly.